### DISCUSSION TITLE
Best Practices for Uploading Large Videos (Server side directly to Blob Storage)

### DISCUSSION THREAD

**Discussion by VirajVaitha123** [vercel/next.js#70078] [Help]
### Summary

Hi,

What's the recommend approach to uploading large videos to blob storage using NextJS 14+ App router.

I have a service that allows users to upload videos from my website, and videos could be up to 1 hour long (over 1GB).

Generally speaking the best approach I believe is:
- to directly upload to blob storage
- Block Upload 
- multi-part form
- SAS Token (equivalent to pre-signed url)
- I want to run this server side as I have a few other operations running with sensitive information, and I feel that server side is best for this video upload.

The current approach I have has a app/upload/route.ts, and the relevant code does the following:

The only way I've found to connect my video upload file, to my server side api is is with :
```
const file = formData.get('file') as File;

await blockBlobClient.uploadData(await file.arrayBuffer(), {
      metadata,
    });
 ```
    
file.arrayBuffer() doesn't seem like the best idea especially for large videos.

Please can anyone advise me on the recommended approach

**Comment by Juneezee** [comment]
> file.arrayBuffer() doesn't seem like the best idea especially for large videos.

You said the uploaded video could be over 1GB. [`.arrayBuffer()` ](https://developer.mozilla.org/en-US/docs/Web/API/Blob/arrayBuffer) reads the entire contents so it is not suitable.

Instead, use [`.stream()`](https://developer.mozilla.org/en-US/docs/Web/API/Blob/stream) to read the uploaded video in chunks.
